server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -283,90 +290,99 @@ private Owner refreshOwner(Owner owner) {
-        Map<String, Content> content = new HashMap<String, Content>();
+        Map<String, ContentData> content = new HashMap<String, ContentData>();
 
         log.info("Refreshing content for {} subscriptions.", subs.size());
 
-        for (Subscription sub : subs) {
-            // Grab all the content from each product
-            this.addProductContentToMap(content, sub.getProduct());
-            this.addProductContentToMap(content, sub.getDerivedProduct());
+        if (subs != null) {
+            for (Subscription sub : subs) {
+                // Grab all the content from each product
+                this.addProductContentToMap(content, sub.getProduct());
+                this.addProductContentToMap(content, sub.getDerivedProduct());
 
-            for (Product product : sub.getProvidedProducts()) {
-                this.addProductContentToMap(content, product);
-            }
+                for (ProductData product : sub.getProvidedProducts()) {
+                    this.addProductContentToMap(content, product);
+                }
 
-            for (Product product : sub.getDerivedProvidedProducts()) {
-                this.addProductContentToMap(content, product);
+                for (ProductData product : sub.getDerivedProvidedProducts()) {
+                    this.addProductContentToMap(content, product);
+                }
             }
         }
 
         Set<Content> changed = this.getChangedContent(owner, content);
 
         // We need to flush here to make sure our pending persists and merges get pushed to the DB
         // so our reference updates don't fail.
-        this.contentCurator.flush();
+        this.ownerContentCurator.flush();
 
         // Go back through each sub and update content references so we don't end up with dangling,
         // transient or duplicate references on any of the subs' products.
-        for (Subscription sub : subs) {
-            this.updateContentRefs(content, owner, sub.getProduct());
-            this.updateContentRefs(content, owner, sub.getDerivedProduct());
+        // for (Subscription sub : subs) {
+        //     this.updateContentRefs(content, owner, sub.getProduct());
+        //     this.updateContentRefs(content, owner, sub.getDerivedProduct());
 
-            for (Product product : sub.getProvidedProducts()) {
-                this.updateContentRefs(content, owner, product);
-            }
+        //     for (Product product : sub.getProvidedProducts()) {
+        //         this.updateContentRefs(content, owner, product);
+        //     }
 
-            for (Product product : sub.getDerivedProvidedProducts()) {
-                this.updateContentRefs(content, owner, product);
-            }
-        }
+        //     for (Product product : sub.getDerivedProvidedProducts()) {
+        //         this.updateContentRefs(content, owner, product);
+        //     }
+        // }
```

Comment:
Yeah. There's a bunch of stuff that's pending removal once it's reviewed and determined to be obsolete/useless. When you see a block like this, it basically means that I was pretty sure removing it is the right thing to do, but not sure enough to delete it entirely. They should be more heavily scrutinized than other parts.